scripts/update.py

```diff
@@ -288,6 +299,101 @@ def format_df_for_chirp(df: pd.DataFrame) -> pd.DataFrame:
 
     return df
 
+def format_df_for_d878(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Format a DataFrame of repeaters for use in Anytone D878.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        All of the repeaters.
+
+    Returns
+    -------
+    pd.DataFrame
+        The FM and NBFM repeaters, and DMR repeaters in the 2m and 70cm bands,
+        formatted for Anytone D878.
+    """
+
+    print("Generating Anytone D878 CSV file.")
+    total_repeaters = len(df.index)
+
+    # Select FM and DMR channels.
+    df = df.loc[df["Mode"].isin(["FM", "NBFM", "DMR"])].copy()
+    numeric_columns = ["Output (MHz)", "Offset (MHz)"]
+    df[numeric_columns] = df[numeric_columns].apply(pd.to_numeric)
+    df_2m = df.loc[(df["Output (MHz)"] > 144.0) & (df["Output (MHz)"] < 148.0)]
+    df_70cm = df.loc[(df["Output (MHz)"] > 430.0) & (df["Output (MHz)"] < 450.0)]
+
+    # Number channels from 1 ...
+    df = pd.concat([df_2m, df_70cm])
+    df.index = list(range(1, len(df) + 1))
+
+    print(f"{len(df.index)} compatible repeaters (out of {total_repeaters} total).")
+
+    df_878 = pd.DataFrame()
+
+    df_878.index.name = "No."
+    df_878["Channel Name"] = df["Callsign"]
+    df_878["Receive Frequency"] = df["Output (MHz)"]
+    df_878["Transmit Frequency"] = df["Output (MHz)"] + df["Offset (MHz)"]
+
+    df_878 = df_878.round(3)
+
+    is_dmr = df["Mode"] == "DMR"
+    df_878.loc[is_dmr, "Channel Type"] = "D-Digital"
+    df_878.loc[~is_dmr, "Channel Type"] = "A-Analog"
+    # Not sure why this seemingly redundant column is in the format?
+    df_878.loc[is_dmr, "DMR MODE"] = "1"
+    df_878.loc[~is_dmr, "DMR MODE"] = "0"
+    
+    # Both DMR and NMFM are "narrow"
+    is_widefm = df["Mode"] == "FM"
+    df_878.loc[is_widefm, "Band Width"] = "25K"
+    df_878.loc[~is_widefm, "Band Width"] = "12.K"
```

Comment:
Is this correct or does this need to be `12.5K` ?

Reply:
Good catch.